### DISCUSSION TITLE
Cant access haspone

### DISCUSSION THREAD

**Discussion by duerrt** [HASwitchPlate/HASPone#184] [Troubleshooting] [answered]
Hi,
I bought the pre-assembled haspone and I generally had it working. Although quite a few
of the blueprints for home assistant didn't work but that's the least of my problems.

Anyways, I decided to move the haspone to a different vlan/wifi on my network. So I did
a factory reset via mqtt and then moved to the other vlan/wifi. I must of screwed up the config
because the haspone is now displaying that its connected to the new vlan/wifi and the 
broker is running on 192.168.30.6. 192.168.30.6 is the address of the haspone. 
I dont think there is a broker running on the haspone, if there is I cant connect to it.
I'm getting a connection refused.

I guess my only recourse is to reflash the device?
I'm looking at this doc and I'm a little confused by this statement.
https://github.com/HASwitchPlate/HASPone/wiki/Flashing-HASPone-to-an-ESP8266

"WeMos D1 mini for HASPone plugged into an available USB port" 

Does my haspone have a usb mini connector on it or do I need to wire one in?

Thanks,
Tom

**Comment by aderusha** [comment]
There is a USB port inside the device which is a "last resort" sort of thing in the event that the device is entirely unresponsive.  Can you bring up the web administration page on the device itself?  Point a browser at the IP shown on the screen and you can perform a factory reset there if that's your goal, and that'll be as good as the USB flash method.

If that doesn't fix things for you, can you post a screenshot of your configuration from that webpage?

**Reply by duerrt** [reply]
Thanks for the quick reply. 
Good to know about the USB port. I figured there must be one but want to double check.

I can't get the admin page to render. Tried multiple browsers and incognito windows. I don't see any JS errors within developer tools. 
I can curl the endpoint and receive a 200 response with what looks like JS for the home page.  Kind of weird. Maybe I can try to do a POST to the endpoint with a "factoryreset" payload?

Does the haspone require internet access?  I've moved the haspone to a vlan that blocks internet access. 

Thanks,
Tom

**Reply by aderusha** [reply]
On bootup and then every 12 hours later it'll check github for an update, but will just ignore that if it doesn't work.  That's the only internet access the device will attempt, so there's no problem running on an isolated VLAN.

For a factory reset, you can issue an HTTP GET to http://[device ip]/resetConfig?confirm=yes

**Reply by duerrt** [reply]
That worked!

Thanks so much for your help!

Tom

**Comment by duerrt** [comment, marked as answer]
Sending a GET to 
`curl http://[device ip]/resetConfig?confirm=yes  `
successfully caused a factory reset
